app/uk/gov/hmrc/transitmovementsrouter/models/MessageTypes.scala

```diff
@@ -22,14 +22,21 @@ import play.api.libs.json.JsSuccess
 import play.api.libs.json.Reads
 import play.api.libs.json.Writes
 
+//TODO: need to take a look at this with respect to whether auditType and movementType needed at this level.
 sealed trait MessageType extends Product with Serializable {
   def code: String
   def rootNode: String
+  def auditType: Option[AuditType] = None
```

Comment:
Better to use this auditType at ```DepartureResponseMessageType``` and ``` ArrivalResponseMessageType``` . As we always need an AuduitType to audit the event so I prefer to make it mandatory.

Reply:
I can see the value in keeping it like this, considering how the audit type is extracted -- it's main draw for me is that it avoids a type match later, which can be expensive (reflection!) compared to a simple dynamic dispatch.

Reply:
The reason I put it at this level because of the the way in which the audit type is extracted, it's cheaper in terms of resources at this level, and it stops it becoming a ticket about tech debt rather than auditing.
That's why the TODO was there: a reminder as to whether we should to take a look at how our message types are structured.